src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -63,7 +63,7 @@ def _validate_deployment_name_with_template_specs(namespace):
             namespace.template_spec = namespace.template_spec.strip("\"")
             if not is_valid_resource_id(namespace.template_spec):
                 raise CLIError('--template-spec is not a valid resource ID.')
-            if namespace.template_spec.__contains__("versions") is False:
+            if (namespace.template_spec in "versions") is False:
```

Comment:
`if not ...`

Reply:
This variable is also reversed, it should be `"versions" in namespace.template_spec`.

Reported to https://github.com/astral-sh/ruff/issues/14423